### ISSUE TITLE
Unable to skopeo copy from multi platform oci-archive to registry.

### ISSUE THREAD

**Issue by kaovilai** [containers/skopeo#2463]
> Came from https://github.com/paketo-buildpacks/builder-jammy-tiny/issues/425
> 
> ```
> ❯ skopeo copy --insecure-policy -a oci-archive:/Users/tiger/git/jammy-tiny-stack/build/run.oci docker://ttl.sh/tiny-jammy:1h
> FATA[0000] initializing source oci-archive:/Users/tiger/git/jammy-tiny-stack/build/run.oci:: more than one image in oci, choose an image 
> ```
> 
> dunno way out here..
> the archive is multi-platform IIUC. 

 _Originally posted by @kaovilai in [#1136](https://github.com/containers/skopeo/issues/1136#issuecomment-2489999673)_

**Comment by mtrmac** [comment]
Thanks for reaching out.

Can you paste the full contents of the `index.json` file at the root of the archive, maybe formatted with `python3 -mjson.tool`, please?

The `oci-archive:`/`oci:` transports only support _some_ shapes of indices / manifests — as a guess, a multi-platform image directly stored in the OCI format as a top-level `index` is not supported; it needs to be an index which contains a nested multi-platform image index.

**Comment by kaovilai** [comment]
index.json from `build/run.oci` generated from `./scripts/create.sh` in [above](https://github.com/containers/skopeo/issues/2463#issue-2679812617) repo
```json
{
   "schemaVersion": 2,
   "mediaType": "application/vnd.oci.image.index.v1+json",
   "manifests": [
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 755,
         "digest": "sha256:7e3d4631e9c027d3eb5083b1d261a40235caf7470565d08a741e4072e048fa85",
         "platform": {
            "architecture": "amd64",
            "os": "linux"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 755,
         "digest": "sha256:4c58342e4e45fe40e4a51112d497b92fc5af9853af0af2fd4457fbb59d263938",
         "platform": {
            "architecture": "arm64",
            "os": "linux"
         }
      }
   ]
}
```

**Comment by kaovilai** [comment]
> is not supported

not working yet, but never will, or not now but maybe?

**Comment by kaovilai** [comment]
If I am presented with an oci archive from a build, do you have any suggestions on how to get it running on macOS?

tl;dr: my root issue is I am unable to run this oci archive with `docker run` or `podman run` on macos
https://github.com/containers/podman/issues/24634

And one way is to skopeo copy to remote registry in a way that will allow `podman run` to work.

**Comment by mtrmac** [comment]
>  as a guess, a multi-platform image directly stored in the OCI format as a top-level index is not supported

For the record, yes, this is the case here.

**Comment by kaovilai** [comment]
> I’d strongly recommend using registries over plain-file archives

[not my choice](https://github.com/paketo-buildpacks/jammy-tiny-stack/blob/main/scripts/create.sh#L124-L125), it's not meant to be used this way tho. these archive would eventually push to https://hub.docker.com/r/paketobuildpacks/run

**Comment by kaovilai** [comment]
Seems like oci archive is [required by jam](https://github.com/paketo-buildpacks/jam/blob/main/commands/create_stack.go#L40) CLI, which they used in the script. I think it's not going be an easy suggestion to purely rely on registries.

**Comment by kaovilai** [comment]
paketobuildpacks are part of the linux foundation with [over 10M docker pulls](https://hub.docker.com/r/paketobuildpacks/run#:~:text=10M%2B,Tags), so I hope that it's considered [widely-available software](https://github.com/containers/skopeo/issues/2463#issuecomment-2492350900)

**Comment by kaovilai** [comment]
Looks like `docker import` command works.. and I was able to supposedly push something to ttl.sh

**Comment by kaovilai** [comment]
Unstale

**Comment by michaelbeaumont** [comment]
I'm running into this issue with tarballs created by `apko build`. It doesn't seem to work with `docker-archive:` either
